plugins/typescript/src/core/schemaToTypeAliasDeclaration.ts

```diff
@@ -239,7 +236,13 @@ export const getType = (
 
       const members: ts.TypeElement[] = Object.entries(
         schema.properties || {}
-      ).map(([key, property]) => {
+      ).map(([key, propertyOrRef]) => {
+        const property = isReferenceObject(propertyOrRef)
+          ? getReferenceSchema(
+              propertyOrRef.$ref,
+              context.openAPIDocument.components
+            )
+          : propertyOrRef;
```

Comment:
Let’s add a unit test for this usecase, so far this was just to have the types compiling.